File: run_modeling.R
args <- commandArgs(trailingOnly = TRUE)

get_arg <- function(name, default = NULL) {
  pattern <- paste0("--", name, "=")
  hit <- args[grepl(pattern, args)]
  if (length(hit) == 0) return(default)
  sub(pattern, "", hit[1])
}

source("R/modeling_module.R")

data_path <- get_arg("data", "data.xlsx")
out_dir <- get_arg("out", "artifacts")
assess <- as.integer(get_arg("assess", "12"))

result <- train_and_evaluate_models(data_path = data_path, output_dir = out_dir, assess = assess)
print(result$evaluation)
cat(sprintf("Modelo seleccionado: %s\n", result$metadata$selected_model))

File: run_modeling.py
from __future__ import annotations

import argparse

from imae_forecasting.modeling import train_and_evaluate_models


def main() -> None:
    parser = argparse.ArgumentParser(description="Construcción y evaluación de modelos para IMAE")
    parser.add_argument("--data", default="data.xlsx", help="Ruta al archivo Excel de entrada")
    parser.add_argument("--out", default="artifacts", help="Directorio de salida")
    parser.add_argument("--splits", default=5, type=int, help="Número de cortes TimeSeriesSplit")
    args = parser.parse_args()

    result = train_and_evaluate_models(data_path=args.data, output_dir=args.out, n_splits=args.splits)
    print("Modelo seleccionado:", result["evaluation"].iloc[0]["model"])
    print("Métricas guardadas en:", f"{args.out}/model_evaluation.csv")


if __name__ == "__main__":
    main()
